packages/twenty-front/src/modules/ui/navigation/menu-item/components/MenuItemDraggable.tsx

```diff
@@ -7,14 +7,15 @@ import { StyledHoverableMenuItemBase } from '../internals/components/StyledMenuI
 import { MenuItemAccent } from '../types/MenuItemAccent';
 
 import { MenuItemIconButton } from './MenuItem';
+import { ReactNode } from 'react';
 
 export type MenuItemDraggableProps = {
-  LeftIcon: IconComponent | undefined;
+  LeftIcon?: IconComponent | undefined;
   accent?: MenuItemAccent;
   iconButtons?: MenuItemIconButton[];
   isTooltipOpen?: boolean;
   onClick?: () => void;
-  text: string;
+  text: ReactNode;
```

Comment:
Could we find another way ? I think text should remain a string type for coherence purpose and if we want to pass down a component, maybe that's because we need to rework this generic UI component ?

Reply:
Actually I think the way is fine, we have other components in the code base using ReactNode type for a `text` entry and we can put `ReactNode` as children of a text component, so it seems legit don't you think ?

Reply:
No, for me we do this for lack of a better component break down, which should be seen with @Bonapara in the design system. That's not ideal. But let's keep it like that as it's not a big problem for now.